docs/developers-guide/devenv.md

```diff
@@ -1,34 +1,54 @@
-# Development Environment
+# Development environment
 
-If you plan to work on the Metabase code and make changes then you'll need to understand a few more things.
+The Metabase application has two basic components:
 
-## Overview
+1. A backend written in Clojure which contains a REST API as well as all the relevant code for talking to databases and processing queries.
+2. A frontend written as a Javascript single-page application which provides the web UI.
 
-The Metabase application has two basic components:
+Both components are built and assembled together into a single JAR file. In the directory where you run the JAR, you can create a JAR file (if Metabase hasn't already created it) and add drivers in there (the drivers are also JARs).
 
-1. a backend written in Clojure which contains a REST API as well as all the relevant code for talking to databases and processing queries.
-2. a frontend written as a Javascript single-page application which provides the web UI.
+## Quick start
 
-Both components are built and assembled together into a single jar file which runs the entire application.
+To spin up a development environment, you'll need to start two terminal sessions: one for the [frontend](#frontend) and one for the [backend](#backend).
 
-## Third-party dependencies
+### Frontend
 
-Metabase depends on lots of third-party libraries to run, so you'll need to keep those up to date. The Clojure CLI will automatically fetch the dependencies when needed. With JavaScript dependencies, however, you'll need to kick off the installation process manually.
+Metabase depends on third-party libraries to run, so you'll need to keep those up to date. The Clojure CLI will automatically fetch the dependencies when needed. With JavaScript dependencies, however, you'll need to kick off the installation process manually.
 
 ```sh
 # javascript dependencies
 $ yarn
 ```
 
-## Development server (quick start)
+Start the frontend build process with
+
+```
+yarn build-hot
+```
+
+See [Frontend development](#frontend-development).
+
+### Backend
+
+Prep dependencies by changing into the Metabase directory and running:
+
+```bash
+clojure -X:deps prep
+cd modules/drivers
+clojure -X:deps prep
+cd ../..
+```
+
+You must run these commands before running any other `clojure` commands in the repository, but you only need to run the prep commands once.
```

Comment:
```suggestion
```

The prep steps aren't needed any more in `master` as of #20611 (merged in yesterday)